src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -471,16 +514,153 @@ private void sendUpdateSingleCaseECC(String authToken, CaseDetails currentCaseDe
             throw new CaseCreationException(MESSAGE + caseIdToLink + e.getMessage());
         }
     }
 
+    /**
+     * Calls reference data API to add HMCTSServiceId to supplementary_data to a case.
+     *
+     * @param caseDetails Details on the case
+     */
+    public void setHmctsServiceIdSupplementary(CaseDetails caseDetails) throws IOException {
+        if (featureToggleService.isGlobalSearchEnabled()) {
+            Map<String, Map<String, Object>> payloadData = Maps.newHashMap();
+            payloadData.put("$set", singletonMap(HMCTS_SERVICE_ID, hmctsServiceId));
+
+            setSupplementaryData(caseDetails, payloadData);
+        }
+    }
+
+    /**
+     * Calls reference data API to add HMCTSServiceId to supplementary_data to a case.
+     *
+     * @param caseDetails Details on the case
+     */
+    public void removeHmctsServiceIdSupplementary(CaseDetails caseDetails) throws IOException {
+        if (featureToggleService.isGlobalSearchEnabled()) {
+            Map<String, Map<String, Object>> payloadData = Maps.newHashMap();
+            payloadData.put("$set", Map.of());
+
+            setSupplementaryData(caseDetails, payloadData);
+        }
+    }
+
+    private void setSupplementaryData(CaseDetails caseDetails,
+                                      Map<String, Map<String, Object>> payloadData) throws IOException {
+        Map<String, Object> payload = Maps.newHashMap();
+        payload.put("supplementary_data_updates", payloadData);
+        String errorMessage = String.format("Call to Supplementary Data API failed for %s",
+                caseDetails.getCaseId());
+
+        try {
+            String adminUserToken = adminUserService.getAdminUserToken();
+            ResponseEntity<Object> response =
+                    ccdClient.setSupplementaryData(adminUserToken, payload, caseDetails.getCaseId());
+            if (response == null) {
+                throw new CaseCreationException(errorMessage);
+            }
+            log.info("Http status received from CCD supplementary update API; {}", response.getStatusCodeValue());
+        } catch (RestClientResponseException e) {
+            throw new CaseCreationException(String.format("%s with %s", errorMessage, e.getMessage()));
+        }
+    }
+
+    private void setCaseNameHmctsInternal(CaseData caseData) {
+        if (caseData.getClaimant() == null) {
+            claimantDefaults(caseData);
+        }
+
+        if (caseData.getRespondent() == null) {
+            respondentDefaults(caseData);
+        }
+
+        caseData.setCaseNameHmctsInternal(caseData.getClaimant() + " vs " + caseData.getRespondent());
+    }
+
+    private void setCaseManagementCategory(CaseData caseData) {
+        caseData.setCaseManagementCategory(DynamicFixedListType.from("Employment Tribunals", "Employment", true));
+    }
+
+    public void setSearchCriteria(CaseData caseData) {
+        if (featureToggleService.isGlobalSearchEnabled()) {
```

Comment:
feature toggle guard clause plz (this is what I meant earlier with larger methods benefitting more from guard clauses, this is a lot of code to have inside an if branch)